v2/devdoc/log_sink_etw_requirements.md

```diff
@@ -20,160 +20,184 @@ The signature of `log_sink_etw.log_sink_log` is:
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint64_t`, the event data descriptor shall be filled with the value of the property by calling `EventDataDescCreate`.
+  - **SRS_LOG_SINK_ETW_01_056: [** If the property type is `LOG_CONTEXT_PROPERTY_TYPE_struct`, a byte with the value `_TlgInSTRUCT | _TlgInChain` shall be added in the metadata. **]**
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_int32_t`, the event data descriptor shall be filled with the value of the property by calling `EventDataDescCreate`.
+  - **SRS_LOG_SINK_ETW_01_081: [** If the property type is any other value, no property data shall be added to the event. **]**
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint32_t`, the event data descriptor shall be filled with the value of the property by calling `EventDataDescCreate`.
+- **SRS_LOG_SINK_ETW_01_057: [** If the property is a struct, an extra byte shall be added in the metadata containing the number of fields in the structure. **]**
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_int16_t`, the event data descriptor shall be filled with the value of the property by calling `EventDataDescCreate`.
+**SRS_LOG_SINK_ETW_01_039: [** `log_sink_etw.log_sink_log` shall fill an `EVENT_DATA_DESCRIPTOR` array of size `2 + 1 + 1 + 1 + 1 + property count`. **]**
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint16_t`, the event data descriptor shall be filled with the value of the property by calling `EventDataDescCreate`.
+Note: 2 entries are for the event descriptor and metadata respectively, 4 entries for the common fields (content, file, func, line) and the rest are 1 for each of the properties.
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_int8_t`, the event data descriptor shall be filled with the value of the property by calling `EventDataDescCreate`.
+**SRS_LOG_SINK_ETW_01_040: [** `log_sink_etw.log_sink_log` shall set event data descriptor at index 2 by calling `_tlgCreate1Sz_char` with the value of the formatted message as obtained by using `printf` with the messages format `message_format` and the arguments in `...`. **]**
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint8_t`, the event data descriptor shall be filled with the value of the property by calling `EventDataDescCreate`.
+**SRS_LOG_SINK_ETW_01_058: [** `log_sink_etw.log_sink_log` shall set event data descriptor at index 3 by calling `_tlgCreate1Sz_char` with `file`. **]**
 
-- If the property type is `LOG_CONTEXT_PROPERTY_TYPE_struct`, no event data descriptor shall be used.
+**SRS_LOG_SINK_ETW_01_059: [** `log_sink_etw.log_sink_log` shall set event data descriptor at index 4 by calling `_tlgCreate1Sz_char` with `func`. **]**
 
-- If the property type is any other value, it shall be ignored.
+**SRS_LOG_SINK_ETW_01_060: [** `log_sink_etw.log_sink_log` shall set event data descriptor at index 5 by calling `EventDataDescCreate` with `line`. **]**
```

Comment:
>EventDataDescCreate [](http://example.com/codeflow?start=114&length=19)

Why is this different than 01_059 or 01_058?

Reply:
Beacuse it is a diffferent type.